packages/frontend/src/components/DataViz/hooks/useChartViz.ts

```diff
@@ -61,8 +67,13 @@ export const useChartViz = <T extends ResultsRunner>({
         if (!config) return undefined;
         if (isVizTableConfig(config)) return undefined;
 
-        return [projectUuid, limit, JSON.stringify(config.fieldConfig)];
-    }, [projectUuid, limit, config]);
+        return [
+            projectUuid,
+            limit,
+            JSON.stringify(config.fieldConfig),
+            additionalQueryKey,
```

Comment:
I would probably destructure this here since additionalQueryKey should be array of items for the query:

```suggestion
            ...(additionalQueryKey ?? []),
```